rest/data/patch.go

```diff
@@ -197,35 +197,56 @@ func AbortPatchesFromPullRequest(ctx context.Context, event *github.PullRequestE
 	return nil
 }
 
+type RawPatch struct {
+	Patch      RawModule   `json:"patch"`
+	RawModules []RawModule `json:"raw_modules"`
+}
+type RawModule struct {
+	Name    string `json:"name"`
+	Diff    string `json:"diff"`
+	Githash string `json:"githash"`
+}
+
 // GetPatchRawPatches fetches the raw patches for a patch
-func GetPatchRawPatches(patchID string) (map[string]string, error) {
+func GetPatchRawPatches(patchID string) (RawPatch, error) {
```

Comment:
nit: seems like returning a pointer is more idiomatic with our code base so that we can return nil in the case of error (same with the other functions in this PR).